### ISSUE TITLE
OSX 10.9.5 - Chef never successfully completed

### ISSUE THREAD

**Issue by sabha** [ptwobrussell/Mining-the-Social-Web-2nd-Edition#250]
Kindly help resolving this issue  "Chef never successfully completed."
```
==> default: [2015-01-15T20:28:06+00:00] DEBUG: running: diff -u /var/chef/cache/get-pip.py /tmp/chef-rest20150115-1544-1obi1h6
==> default: [2015-01-15T20:28:06+00:00] DEBUG: reading modes from /var/chef/cache/get-pip.py file
==> default: [2015-01-15T20:28:06+00:00] DEBUG: applying mode = 644, uid = 0, gid = 0 to /tmp/chef-rest20150115-1544-1obi1h6
==> default: [2015-01-15T20:28:06+00:00] DEBUG: moving temporary file /tmp/chef-rest20150115-1544-1obi1h6 into place at /var/chef/cache/get-pip.py
==> default: [2015-01-15T20:28:06+00:00] INFO: remote_file[/var/chef/cache/get-pip.py] updated file contents /var/chef/cache/get-pip.py
==> default: [2015-01-15T20:28:06+00:00] DEBUG: found current_mode == nil, so we are creating a new file, updating mode
==> default: [2015-01-15T20:28:06+00:00] DEBUG: found current_mode == nil, so we are creating a new file, updating mode
==> default: [2015-01-15T20:28:06+00:00] DEBUG: found target_uid == nil, so no owner was specified on resource, not managing owner
==> default: [2015-01-15T20:28:06+00:00] DEBUG: found target_gid == nil, so no group was specified on resource, not managing group
==> default: [2015-01-15T20:28:06+00:00] DEBUG: found target_uid == nil, so no owner was specified on resource, not managing owner
==> default: [2015-01-15T20:28:06+00:00] DEBUG: found target_gid == nil, so no group was specified on resource, not managing group
==> default: [2015-01-15T20:28:06+00:00] DEBUG: found current_mode == nil, so we are creating a new file, updating mode
==> default: [2015-01-15T20:28:06+00:00] INFO: remote_file[/var/chef/cache/get-pip.py] mode changed to 644
==> default: [2015-01-15T20:28:06+00:00] DEBUG: selinux utilities can not be found. Skipping selinux permission fixup.
==> default: [2015-01-15T20:28:06+00:00] INFO: Processing execute[install-setuptools] action run (python::pip line 48)
==> default: [2015-01-15T20:28:06+00:00] INFO: Processing execute[Guard resource] action run (dynamically defined)
==> default: [2015-01-15T20:28:06+00:00] DEBUG: providers for generic execute resource enabled on node include: [Chef::Provider::Execute]
==> default: [2015-01-15T20:28:06+00:00] DEBUG: providers that refused resource execute[Guard resource] were: []
==> default: [2015-01-15T20:28:06+00:00] DEBUG: providers that support resource execute[Guard resource] include: [Chef::Provider::Execute]
==> default: [2015-01-15T20:28:06+00:00] DEBUG: providers that survived replacement include: [Chef::Provider::Execute]
==> default: [2015-01-15T20:28:06+00:00] DEBUG: providers for generic execute resource enabled on node include: [Chef::Provider::Execute]
==> default: [2015-01-15T20:28:06+00:00] DEBUG: providers that refused resource execute[install-setuptools] were: []
==> default: [2015-01-15T20:28:06+00:00] DEBUG: providers that support resource execute[install-setuptools] include: [Chef::Provider::Execute]
==> default: [2015-01-15T20:28:06+00:00] DEBUG: providers that survived replacement include: [Chef::Provider::Execute]
==> default: 
==> default: ================================================================================
==> default: Error executing action `run` on resource 'execute[install-setuptools]'
==> default: ================================================================================
==> default: 
==> default: 
==> default: Mixlib::ShellOut::ShellCommandFailed
==> default: ------------------------------------
==> default: Expected process to exit with [0], but received '127'
==> default: 
==> default: ---- Begin output of   /bin/python ez_setup.py
==> default: 
==> default:  ----
==> default: STDOUT: 
==> default: 
==> default: STDERR: sh: 1: /bin/python: not found
==> default: 
==> default: ---- End output of   /bin/python ez_setup.py
==> default:  ----
==> default: Ran   /bin/python ez_setup.py
==> default:  returned 127
==> default: 
==> default: Resource Declaration:
==> default: ---------------------
==> default: # In /tmp/vagrant-chef/4169e2c238121dd8f02adc312902b329/cookbooks/python/recipes/pip.rb
==> default: 
==> default:  48: execute "install-setuptools" do
==> default:  49:   cwd Chef::Config[:file_cache_path]
==> default:  50:   command <<-EOF
==> default:  51:   #{node['python']['binary']} ez_setup.py
==> default:  52:   EOF
==> default: 
==> default:  53:   not_if "#{node['python']['binary']} -c 'import setuptools'"
==> default: 
==> default:  54: end
==> default:  55: 
==> default: 
==> default: 
==> default: Compiled Resource:
==> default: ------------------
==> default: # Declared in /tmp/vagrant-chef/4169e2c238121dd8f02adc312902b329/cookbooks/python/recipes/pip.rb:48:in `from_file'
==> default: 
==> default: 
==> default: 
==> default: execute("install-setuptools") do
==> default: 
==> default:   action "run"
==> default: 
==> default:   retries 0
==> default: 
==> default:   retry_delay 2
==> default:   default_guard_interpreter :execute
==> default: 
==> default:   command "  /bin/python ez_setup.py\n"
==> default: 
==> default:   backup 5
==> default: 
==> default:   cwd "/var/chef/cache"
==> default: 
==> default:   returns 0
==> default: 
==> default:   declared_type :execute
==> default:   cookbook_name :python
==> default:   recipe_name "pip"
==> default:   not_if "/bin/python -c 'import setuptools'"
==> default: end
==> default: 
==> default: 
==> default: 
==> default: [2015-01-15T20:28:07+00:00] INFO: Running queued delayed notifications before re-raising exception
==> default: [2015-01-15T20:28:07+00:00] DEBUG: Re-raising exception: Mixlib::ShellOut::ShellCommandFailed - execute[install-setuptools] (python::pip line 48) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '127'
==> default: ---- Begin output of   /bin/python ez_setup.py
==> default:  ----
==> default: STDOUT: 
==> default: STDERR: sh: 1: /bin/python: not found
==> default: ---- End output of   /bin/python ez_setup.py
==> default:  ----
==> default: Ran   /bin/python ez_setup.py
==> default:  returned 127
==> default: /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/mixlib-shellout-2.0.0/lib/mixlib/shellout.rb:267:in `invalid!'
==> default:   /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/mixlib-shellout-2.0.0/lib/mixlib/shellout.rb:254:in `error!'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/mixin/shell_out.rb:50:in `shell_out!'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/provider/execute.rb:60:in `block in action_run'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/mixin/why_run.rb:52:in `call'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/mixin/why_run.rb:52:in `add_action'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/provider.rb:180:in `converge_by'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/provider/execute.rb:59:in `action_run'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/provider.rb:145:in `run_action'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/resource.rb:582:in `run_action'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/runner.rb:49:in `run_action'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `block (2 levels) in converge'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `each'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `block in converge'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/resource_collection/resource_list.rb:83:in `block in execute_each_resource'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:116:in `call'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:116:in `call_iterator_block'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:85:in `step'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:104:in `iterate'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:55:in `each_with_index'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/resource_collection/resource_list.rb:81:in `execute_each_resource'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/runner.rb:80:in `converge'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/client.rb:315:in `converge'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/client.rb:400:in `block in run'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/client.rb:399:in `catch'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/client.rb:399:in `run'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/application.rb:261:in `block in fork_chef_client'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/application.rb:249:in `fork'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/application.rb:249:in `fork_chef_client'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/application.rb:215:in `block in run_chef_client'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/local_mode.rb:38:in `with_server_connectivity'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/application.rb:201:in `run_chef_client'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/application/solo.rb:245:in `block in interval_run_chef_client'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/application/solo.rb:234:in `loop'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/application/solo.rb:234:in `interval_run_chef_client'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/application/solo.rb:224:in `run_application'
==> default:   /opt/chef/embedded/apps/chef/lib/chef/application.rb:58:in `run'
==> default:   /opt/chef/embedded/apps/chef/bin/chef-solo:25:in `<top (required)>'
==> default:   /usr/bin/chef-solo:40:in `load'
==> default:   /usr/bin/chef-solo:40:in `<main>'
==> default: [2015-01-15T20:28:07+00:00] ERROR: Running exception handlers
==> default: [2015-01-15T20:28:07+00:00] ERROR: Exception handlers complete
==> default: [2015-01-15T20:28:07+00:00] FATAL: Stacktrace dumped to /var/chef/cache/chef-stacktrace.out
==> default: [2015-01-15T20:28:07+00:00] DEBUG: Mixlib::ShellOut::ShellCommandFailed: execute[install-setuptools] (python::pip line 48) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '127'
==> default: ---- Begin output of   /bin/python ez_setup.py
==> default:  ----
==> default: STDOUT: 
==> default: STDERR: sh: 1: /bin/python: not found
==> default: ---- End output of   /bin/python ez_setup.py
==> default:  ----
==> default: Ran   /bin/python ez_setup.py
==> default:  returned 127
==> default: /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/mixlib-shellout-2.0.0/lib/mixlib/shellout.rb:267:in `invalid!'
==> default: /opt/chef/embedded/lib/ruby/gems/2.1.0/gems/mixlib-shellout-2.0.0/lib/mixlib/shellout.rb:254:in `error!'
==> default: /opt/chef/embedded/apps/chef/lib/chef/mixin/shell_out.rb:50:in `shell_out!'
==> default: /opt/chef/embedded/apps/chef/lib/chef/provider/execute.rb:60:in `block in action_run'
==> default: /opt/chef/embedded/apps/chef/lib/chef/mixin/why_run.rb:52:in `call'
==> default: /opt/chef/embedded/apps/chef/lib/chef/mixin/why_run.rb:52:in `add_action'
==> default: /opt/chef/embedded/apps/chef/lib/chef/provider.rb:180:in `converge_by'
==> default: /opt/chef/embedded/apps/chef/lib/chef/provider/execute.rb:59:in `action_run'
==> default: /opt/chef/embedded/apps/chef/lib/chef/provider.rb:145:in `run_action'
==> default: /opt/chef/embedded/apps/chef/lib/chef/resource.rb:582:in `run_action'
==> default: /opt/chef/embedded/apps/chef/lib/chef/runner.rb:49:in `run_action'
==> default: /opt/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `block (2 levels) in converge'
==> default: /opt/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `each'
==> default: /opt/chef/embedded/apps/chef/lib/chef/runner.rb:81:in `block in converge'
==> default: /opt/chef/embedded/apps/chef/lib/chef/resource_collection/resource_list.rb:83:in `block in execute_each_resource'
==> default: /opt/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:116:in `call'
==> default: /opt/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:116:in `call_iterator_block'
==> default: /opt/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:85:in `step'
==> default: /opt/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:104:in `iterate'
==> default: /opt/chef/embedded/apps/chef/lib/chef/resource_collection/stepable_iterator.rb:55:in `each_with_index'
==> default: /opt/chef/embedded/apps/chef/lib/chef/resource_collection/resource_list.rb:81:in `execute_each_resource'
==> default: /opt/chef/embedded/apps/chef/lib/chef/runner.rb:80:in `converge'
==> default: /opt/chef/embedded/apps/chef/lib/chef/client.rb:315:in `converge'
==> default: /opt/chef/embedded/apps/chef/lib/chef/client.rb:400:in `block in run'
==> default: /opt/chef/embedded/apps/chef/lib/chef/client.rb:399:in `catch'
==> default: /opt/chef/embedded/apps/chef/lib/chef/client.rb:399:in `run'
==> default: /opt/chef/embedded/apps/chef/lib/chef/application.rb:261:in `block in fork_chef_client'
==> default: /opt/chef/embedded/apps/chef/lib/chef/application.rb:249:in `fork'
==> default: /opt/chef/embedded/apps/chef/lib/chef/application.rb:249:in `fork_chef_client'
==> default: /opt/chef/embedded/apps/chef/lib/chef/application.rb:215:in `block in run_chef_client'
==> default: /opt/chef/embedded/apps/chef/lib/chef/local_mode.rb:38:in `with_server_connectivity'
==> default: /opt/chef/embedded/apps/chef/lib/chef/application.rb:201:in `run_chef_client'
==> default: /opt/chef/embedded/apps/chef/lib/chef/application/solo.rb:245:in `block in interval_run_chef_client'
==> default: /opt/chef/embedded/apps/chef/lib/chef/application/solo.rb:234:in `loop'
==> default: /opt/chef/embedded/apps/chef/lib/chef/application/solo.rb:234:in `interval_run_chef_client'
==> default: /opt/chef/embedded/apps/chef/lib/chef/application/solo.rb:224:in `run_application'
==> default: /opt/chef/embedded/apps/chef/lib/chef/application.rb:58:in `run'
==> default: /opt/chef/embedded/apps/chef/bin/chef-solo:25:in `<top (required)>'
==> default: /usr/bin/chef-solo:40:in `load'
==> default: /usr/bin/chef-solo:40:in `<main>'
==> default: [2015-01-15T20:28:08+00:00] ERROR: execute[install-setuptools] (python::pip line 48) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '127'
==> default: ---- Begin output of   /bin/python ez_setup.py
==> default:  ----
==> default: STDOUT: 
==> default: STDERR: sh: 1: /bin/python: not found
==> default: ---- End output of   /bin/python ez_setup.py
==> default:  ----
==> default: Ran   /bin/python ez_setup.py
==> default:  returned 127
==> default: [2015-01-15T20:28:08+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
Chef never successfully completed! Any errors should be visible in the
output above. Please fix your recipes so that they properly complete.
```

**Comment by AndyMC88** [comment]
did you discover how make it work ?

**Comment by ptwobrussell** [comment]
See #243

**Comment by AndyMC88** [comment]
I can´t make it work in windows 7 I will keep trying...   lucky enough also have ubuntu 14.04 where I follow the instruction as in the video an run beautiful. I´m starting to prefer ubuntu. Thanks !!

**Comment by sabha** [comment]
#1 : Error executing action `run` on resource 'execute[install-setuptools]'
@AndyMC88 : Sorry to hear , to resolve this error 'execute[install-setuptools]' , I followed these 4 steps. Later I didn't get that error so closed it , but now I am facing another error ('mongodb-10gen') ,

vagrant ssh
sudo ln -s /usr/bin/python /bin/python
vagrant halt
vagrant up --provision

#2 : Error executing action `install` on resource 'apt_package[mongodb-10gen]'
@ptwobrussell : Now facing this issues 'apt_package[mongodb-10gen]' , you have mentioned in some other post that it is a network issues. I have tried several time , 
vagrant destroy
vagrant up

Once mount is happened opened another terminal and ran this to avoid 1st issue ,
vagrant ssh
sudo ln -s /usr/bin/python /bin/python

Need help resolving 'apt_package[mongodb-10gen]' issues.

**Comment by AndyMC88** [comment]
I´m not an expert but it looks like you need to dowload this "mongodb", give it a try.

**Comment by sabha** [comment]
Confused now , when vagrant is going do SSH and install necessary software via Chef (configuration management) then why should I have to install/link few things manually. 
Curious to know is there another working box besides 'precise64' available for MAC OSX 10.9.5 along with latest Vagrant & Virtualbox?

**Comment by sabha** [comment]
Resolved successfully by installing MongoDB via vagrant ssh , http://docs.mongodb.org/manual/tutorial/install-mongodb-on-ubuntu/. Thanks @AndyMC88 it works . .
